src/basins/basins_utilities.jl

```diff
@@ -86,7 +86,11 @@ function convergence_and_basins_of_attraction(mapper::AttractorMapper, grid; sho
         @error "The mapper and the grid must have the same dimension"
     end
     basins = zeros(length.(grid))
-    iterations = zeros(Int, length.(grid))
+    if isa(mapper.ds,ContinuousDynamicalSystem)
+        iterations = zeros(length.(grid))
+    else
+        iterations = zeros(Int, length.(grid))
+    end
```

Comment:
Instead of doing all of this we can do `zeros(typeof(current_time(ds)), ...)`. No need for Ifs.

Also, please don't use `mapper.ds`. Use instead `referenced_dynamical_system(mapper)`.